xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/PVR.h

```diff
@@ -0,0 +1,7939 @@
+  ///       addonInstance = new CMyPVRClientInstance_Type2(instance, version);
+  ///     return ADDON_STATUS_OK;
+  ///   }
+  ///   else if (...)
+  ///   {
+  ///     ...
+  ///   }
+  ///   return ADDON_STATUS_UNKNOWN;
+  /// }
+  ///
+  /// ...
+  /// ~~~~~~~~~~~~~
+  ///
+  #define PVR_STREAM_PROPERTY_INPUTSTREAM_INSTANCE_ID STREAM_PROPERTY_INPUTSTREAM_INSTANCE_ID
+
+  /// @brief the MIME type of the stream that should be played.
+  ///
+  #define PVR_STREAM_PROPERTY_MIMETYPE "mimetype"
+
+  /// @brief <b>"true"</b> to denote that the stream that should be played is a
+  /// realtime stream.
+  ///
+  /// Any other value indicates that this is no realtime stream.
+  ///
+  #define PVR_STREAM_PROPERTY_ISREALTIMESTREAM STREAM_PROPERTY_ISREALTIMESTREAM
+
+  /// @brief <b>"true"</b> to denote that if the stream is from an EPG tag.
+  ///
+  /// It should be played is a live stream. Otherwise if it's a EPG tag it will
+  /// play as normal video.
+  ///
+  #define PVR_STREAM_PROPERTY_EPGPLAYBACKASLIVE "epgplaybackaslive"
+
+  /// @brief Special value for @ref PVR_STREAM_PROPERTY_INPUTSTREAM to use
+  /// ffmpeg to directly play a stream URL.
+  #define PVR_STREAM_PROPERTY_VALUE_INPUTSTREAMFFMPEG STREAM_PROPERTY_VALUE_INPUTSTREAMFFMPEG
+
+  //@}
+  //-----------------------------------------------------------------------------
+
+  //============================================================================
+  /// @ingroup cpp_kodi_addon_pvr_Defs_Various
+  /// @brief Denotes that no channel uid is available.
+  ///
+  /// Special @ref kodi::addon::PVRTimer::SetClientChannelUid and
+  /// @ref kodi::addon::PVRRecording::SetChannelUid value to indicate that no
+  /// channel uid is available.
+  #define PVR_CHANNEL_INVALID_UID -1
+  //----------------------------------------------------------------------------
+
+  //}}}
+  //____________________________________________________________________________
+  //--==----==----==----==----==----==----==----==----==----==----==----==----==
+  //¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+  // "C" Internal interface tables for intercommunications between addon and kodi
+  //{{{
+
+  struct AddonInstance_PVR;
+
+  /*!
+   * @brief Structure to define typical standard values
+   */
+  typedef struct AddonProperties_PVR
+  {
+    const char* strUserPath;
+    const char* strClientPath;
+    int iEpgMaxDays;
+  } AddonProperties_PVR;
+
+  /*!
+   * @brief Structure to transfer the methods from Kodi to addon
+   */
+  typedef struct AddonToKodiFuncTable_PVR
+  {
+    // Pointer inside Kodi where used from him to find his class
+    KODI_HANDLE kodiInstance;
+
+    //--==----==----==----==----==----==----==----==----==----==----==----==----==
+    // General callback functions
+    void (*AddMenuHook)(void* kodiInstance, struct PVR_MENUHOOK* hook);
+    void (*Recording)(void* kodiInstance, const char* Name, const char* FileName, bool On);
```

Comment:
"Does C have bool type?

Standard C (since C99) provides a boolean type, called _Bool . By including the header stdbool. h , one can use the more intuitive name bool and the constants true and false ."

SO, either include stdbool.h or use _Bool or replace the booleans in function declarations with int.

Reply:
The `#include <stdbool.h>` is in header, also was the compile test with "C" OK. Thats not complicated for me leave with `bool`, as with them some `aaa != 0 ? return true : return false;` can be prevented.

Reply:
Okay, stdbool.h inclusion is fine by me. Just wanted to make sure you are aware of the bool problem in C. Obviously you are.